extensions/standard-processors/processors/GetTCP.cpp

```diff
@@ -130,7 +130,7 @@ void GetTCP::onSchedule(const std::shared_ptr<core::ProcessContext> &context, co
   }
 
   if (context->getProperty(StayConnected.getName(), value)) {
-    utils::StringUtils::StringToBool(value, stay_connected_);
+    stay_connected_ = utils::StringUtils::toBool(value).value_or(false);
```

Comment:
I would make the default true, because we set `stay_connected_` to true in the `else` branch.

Reply:
Quick question: Do we need else branch here?

Reply:
Ah yes, I can see we have `stay_connected_ = true;` earlier.  You're right, a good option would be to (move that assignment down to just before line 132 and) remove the `else` branch.  [And also change the default to `true`.]

Reply:
And it's initialized to `true` on line 122, which means it used to stay true in the case of an unsuccessful conversion.

Reply:
So my takeaway is that a default value is not really needed here.

Reply:
I think that is not the right conclusion. As Feri said, you should move the default value assignment right before this check and make the "default" argument of `value_or` true.

This should be equivalent but a bit less readable:
```c++
stay_connected_ = (!context->getProperty(StayConnected.getName(), value)) || utils::StringUtils::toBool(value).value_or(true);
```

Reply:
Fixed :)